### PR TITLE
add icon-only variant for Button

### PR DESCRIPTION
**Pull request by jluyau**
Closes 

- adjusts Button styling to be a rounded button if there's only an Icon child

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

Adobe/Quarry

### CODE REVIEW

File: packages/@react-spectrum/button/src/Button.tsx [hunk 1]
```diff
@@ -49,6 +49,15 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
     buttonVariant = VARIANT_MAPPING[variant];
   }
 
+  let hasIconOnlyChild = useMemo(() => {
+    if (React.Children.count(children) !== 1) {
+      return false;
+    }
+    let child = React.Children.toArray(children)[0];
+    // @ts-ignore
+    return React.isValidElement(child) && typeof child.type === 'function' && child.type().type.name === 'Icon';
```
Comment (jluyau) [inline packages/@react-spectrum/button/src/Button.tsx:58]: this is kinda ugly, so if someone has a better way of detecting an `Icon` type, definitely willing to update

**Review by snowystinger — COMMENTED** [review]
Can't we handle this the same way ActionButton handles icon only? shouldn't need that check for hasIconOnlyChild at all

**Comment by jluyau** [comment]
> Can't we handle this the same way ActionButton handles icon only? shouldn't need that check for hasIconOnlyChild at all

That was how I was going to originally approach the problem, but the `<button>` element has a class with a min-width that I would not be able to affect from the icon child, so that's why I need to detect if i have a single Icon element, then apply a secondary class to the button.

**Comment by snowystinger** [comment]
Ah, dang, that's annoying. We use slots in button though, so should be able to use the hasChild function https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/dialog/src/Dialog.tsx#L60 since we can't use the native CSS selector yet (soon fingers crossed)

**Review by devongovett — DISMISSED** [review]

File: packages/@react-spectrum/button/src/Button.tsx [hunk 2]
```diff
@@ -62,6 +71,7 @@ function Button<T extends ElementType = 'button'>(props: SpectrumButtonProps<T>,
             `spectrum-Button--${buttonVariant}`,
             {
               'spectrum-Button--quiet': isQuiet,
+              'spectrum-Button--iconOnly': hasIcon && !hasLabel,
```
Comment (snowystinger) [inline packages/@react-spectrum/button/src/Button.tsx:74]: probably could just use `!hasLabel`
wouldn't make much sense to have a button with no label AND no icon

**Review by snowystinger — APPROVED** [review]
In general looks good, one possible optimization

**Review by snowystinger — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**